From: Tim Dengel <tim.dengel.debian@gmail.com>
Date: Tue, 18 Oct 2016 20:47:59 +0200
Subject: Fix segfault

---
 src/gt-channel.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gt-channel.c b/src/gt-channel.c
index 2fcc016..6aa49cc 100644
--- a/src/gt-channel.c
+++ b/src/gt-channel.c
@@ -137,6 +137,9 @@ update_set_cb(gpointer udata)
     GtChannel* self = GT_CHANNEL(udata);
     GtChannelRawData* raw = g_object_get_data(G_OBJECT(self), "raw-data");
 
+    if(!raw)
+        return FALSE;
+
     g_info("{GtChannel} Finished update '%s'", raw->name);
 
     gt_channel_update_from_raw_data(self, raw);
